packages/frontend/src/pages/Tiles/index.tsx

```diff
@@ -18,55 +20,58 @@ const TILES_TITLE = 'Tiles'
 const TILES_PER_PAGE = 10
 
 export default function Tiles(): JSX.Element {
-  const [searchParams, setSearchParams] = useSearchParams()
-  const page = parseInt(searchParams.get('page') || '', 10) || 1
+  const { input, page, setSearchParams } = usePaginationAndFilter()
 
   const { data, loading: isTileListLoading } = useQuery<{
     getTables: PaginatedTables
   }>(GET_TABLES, {
     variables: {
       limit: TILES_PER_PAGE,
       offset: (page - 1) * TILES_PER_PAGE,
+      name: input,
     },
   })
 
-  if (!data?.getTables || isTileListLoading) {
-    return (
-      <Center h="100%">
-        <PrimarySpinner fontSize="4xl" thickness="4px" margin="auto" />
-      </Center>
-    )
-  }
-
-  const { pageInfo, edges } = data.getTables || {}
+  const { pageInfo, edges } = data?.getTables || { edges: [] }
   const tilesToDisplay = edges.map(({ node }) => node)
-  const isEmpty = tilesToDisplay?.length === 0
+  const hasNoTiles =
+    !isTileListLoading && tilesToDisplay?.length === 0 && input.trim() === ''
+  const hasNoSearchResults =
+    !isTileListLoading && tilesToDisplay?.length === 0 && input.trim() !== ''
```

Comment:
oh why is the `isEmpty = tilesToDisplay?.length === 0` removed

Reply:
🤷🏻